src/ring_theory/localization.lean

```diff
@@ -757,13 +757,14 @@ def lin_coe : R →ₗ[R] f.codomain :=
   map_add'  := f.to_map.map_add,
   map_smul' := f.to_map.map_mul }
 
-variables {f}
+/-- Map from ideals of `R` to submodules of `S` induced by `f`.
+(This was previously a `has_coe` instance, but if `f.codomain = R` then this will loop.) -/
+def coe_submodule (I : ideal R) : submodule R f.codomain := submodule.map f.lin_coe I
```

Comment:
This is also dangerous as a `has_coe_t` instance? How does this relate to the note [[use has_coe_t]](https://leanprover-community.github.io/mathlib_docs/notes.html#use%20has_coe_t)?

What do you exactly mean with this?

> if `f.codomain = R` then this will loop

Reply:
If `f.codomain = R` then both sides are defeq, i.e. it is an identity coercion. `has_coe_t` would help somewhat here, but that's not the only problem with this instance; `f` is not determined by the lhs so instance search goes down a rabbit hole.

Reply:
`localization_map.codomain` is not reducible, so I thought that for the purposes of type-class inference `f.codomain` and `R` are never considered definitionally equal?

If it were a `has_coe_t` instance, I'm sure if the fact that `f` is not determined by the lhs is important...?

Reply:
Changing it to a `has_coe_t` instance is enough to avoid the loop that exposed this, yes.

Reply:
I'm happy to keep it a `def` though, I'm not sure whether this is a good coercion. Maybe a note saying that it doesn't loop as a `has_coe_t` is nice?